doc/string/partition.rdoc

```diff
@@ -1,24 +1,32 @@
 Returns a 3-element array of substrings of +self+.
 
-Matches a pattern against +self+, scanning from the beginning.
-The pattern is:
+Converts +pattern+ to a Regexp, if it is not already one.
```

Comment:
I don't think this is true, it doesn't convert `pattern` to a Regexp. Looking at the code, if `pattern` is not a String or Regexp, it converts it to a String. If it's a String, then it uses `String#index` to get the index at which the match occurs.